app/models/yellow_ribbon_program_source.rb

```diff
@@ -4,48 +4,89 @@ class YellowRibbonProgramSource < ApplicationRecord
+    'initials yr processor' => { column: :initials_yr_processor },
+    'missed deadline' => { column: :missed_deadline, converter: BooleanConverter },
+    'modified' => { column: :modified, converter: BooleanConverter },
     'new school' => { column: :new_school, converter: BooleanConverter },
+    'notes' => { column: :notes },
     'open ended agreement' => { column: :open_ended_agreement, converter: BooleanConverter },
-    'modified' => { column: :modified, converter: BooleanConverter },
-    'withdrawn' => { column: :withdrawn, converter: BooleanConverter },
+    'public private' => { column: :public_private },
+    'school name in weams' => { column: :school_name_in_weams },
+    'sco email address' => { column: :sco_email_address },
     'sco name' => { column: :sco_name },
     'sco telephone number' => { column: :sco_telephone_number },
-    'sco email address' => { column: :sco_email_address },
+    'sfr email address' => { column: :sfr_email_address },
     'sfr name' => { column: :sfr_name },
     'sfr telephone number' => { column: :sfr_telephone_number },
-    'sfr email address' => { column: :sfr_email_address },
-    'initials yr processor' => { column: :initials_yr_processor },
-    'year of yr participation' => { column: :year_of_yr_participation },
-    'notes' => { column: :notes }
+    'updated for 2011-2012' => { column: :updated_for_2011_2012, converter: BooleanConverter },
+    'withdrawn' => { column: :withdrawn, converter: BooleanConverter },
+    'year of yr participation' => { column: :year_of_yr_participation }
   }.freeze
 
-  validates :facility_code, presence: true
+  # validates :city, presence: true
+  validates :contribution_amount, numericality: true
   validates :degree_level, presence: true
   validates :division_professional_school, presence: true
+  validates :facility_code, presence: true
   validates :number_of_students, numericality: true
-  validates :contribution_amount, numericality: true
+  # validates :school_name_in_yr_database, presence: true
+  # validates :state, presence: true
+  # validates :street_address, presence: true
+  # validates :zip, presence: true
+
+  # Finds exact-matching facility_code or partial-matching school and city names
+  scope :search, lambda { |search_term|
+    return if search_term.blank?
+
+    clause = [
+      'lower(school_name_in_yr_database) LIKE (:search_term)'
+    ]
+
+    where(
+      sanitize_sql_for_conditions(
+        [clause.join(' OR '),
+         search_term: "%#{search_term}%"]
+      )
+    )
+  }
+
+  scope :filter, lambda { |field, value|
+    return if value.blank?
+    raise ArgumentError, 'Field name is required' if field.blank?
+
+    case value
+    when 'true', 'yes'
+      where(field => true)
+    when 'false', 'no'
+      where.not(field => true)
+    else
+      where(field => value)
+    end
+  }
+
+  scope :version, ->(n) { where(version: n) }
+
+  scope :filter_count, lambda { |field|
```

Comment:
is this needed since switching to `yellow_ribbon_program`?

Reply:
Nope! Nice catch 😎